### ISSUE TITLE
function-argument does not work if there is  a inline function as a default value

### ISSUE THREAD

**Issue by aonghusonia** [alesmenzel/vargs#1]
```js
const functionArguments = require('@alesmenzel/function-arguments');

functionArguments(
  (a, b = 1, c = Math.square(4), d = () => console.log("h"), alert) => {}
);
// [ 'a', 'b = 1', 'c = Math.square(4)', 'd = (a', 'b' ]

functionArguments(
  function(a, b = 1, c = Math.square(4), d = (a,b) => console.log("h"), alert){}
);

// [ 'a', 'b = 1', 'c = Math.square(4' ]
   
```
https://www.npmjs.com/package/fn-args

looks like it may be a good alternative i wasn't able to break it

**Comment by alesmenzel** [comment]
Hello, thanks for the report, **i am happy to accept a PR that fixes it**, otherwise i am considering using AST parser instead of regex, which would work with those cases.

**Comment by alesmenzel** [comment]
Prepared new version with the AST parser, it is blocked now until this bug is resolved: https://github.com/acornjs/acorn/issues/710

**Comment by alesmenzel** [comment]
Fixed in 1.0.0
